lms/djangoapps/instructor/lti_grader.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python
+"""
+Send grades to an OpenEdX LTI component
+"""
+import csv
+
+from instructor import lti_connection
+from instructor.lti_connection import LTIConnectionError
+
+
+class LTIGrader(object):
+    """
+    Updates grades for a given LTI component
+    """
+
+    def __init__(self, course_id, url_base, lti_key, lti_secret):
+        self.url_base = url_base
+        self.course_id = course_id
+        self.key = lti_key
+        self.secret = lti_secret
+        self.output = {'success': '', 'error': ''}
+
+    def _get_first_anon_id(self, grade_data):
+        """
+        Returns the first anonymized student id for in the given uploaded csv data file
+        :param grade_data: an InMemoryUploadedFile with lti grade data
+        :return: anon_id
+        """
+        reader = self._unicode_csv_reader(grade_data)
+        try:
+            reader.next()
+            first_row = reader.next()
+            anon_id = first_row[1]
+        except StopIteration:
+            self.output['error'] += '\nThe CSV file contains no useful data!'
+            anon_id = None
+        return anon_id
+
+    def update_grades(self, grade_data):
+        """
+        Updates grades for the LTI component with the grade data provided
+        :param grade_data: an InMemoryUploadedFile with lti grade data
+        :return: output, a dictionary with two keys: 'success' and 'error'
+        """
+        grade_data = grade_data.read().splitlines()
```

Comment:
Also the reason I am reading the entire file is because this was because this solved the issue raised by @mondiaz in her last comment re "demo requested feature" in 
https://lagunita.atlassian.net/browse/LE-357?filter=-1